### PR TITLE
Create; initial version of tooltip

### PR DESCRIPTION
**Pull request by chrisopedia**
Why is this pull request necessary:

1. create a CSS only tooltip

Changes proposed in this pull request:

- add in examples to showcase
- mixin to generate css for tooltip
- helper to use mixin

image

### CODE REVIEW

File: styles/helpers/tooltip.css [hunk 1]
```diff
@@ -0,0 +1,137 @@
+button.tooltip {
```
Comment (jgallen23) [inline styles/helpers/tooltip.css]: we shouldn't tie this to button elements

File: styles/helpers/tooltip.css [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  overflow: visible;
+}
+.tooltip {
+  @mixin tooltip;
```
Comment (jgallen23) [inline styles/helpers/tooltip.css]: we should have the mixin handle all the css.  We need to think of mixins as standalone functions that can work by themselves and then the helper classes just use the mixins

File: styles/helpers/tooltip.css [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    &:after {
+      white-space: normal;
+    }
+  }
+
+  &-up {
+    &::before {
+      triangle: equilateral pointing-down;
+      height: 6px;
+      background-color: rgba(17,17,17,.9);
```
Comment (jgallen23) [inline styles/helpers/tooltip.css]: we should never hard code colors

File: styles/helpers/tooltip.css (hunk 1 above)
Comment (chrisopedia) [inline styles/helpers/tooltip.css]: its not.  its a hack for ie11

File: styles/mixins/tooltip.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+module.exports = function () {
+  return function (mixin, color) {
+    var color = color || 'rgba(17,17,17,.9)';
```
Comment (jgallen23) [inline styles/mixins/tooltip.js]: can we read this from config? probably want both background and foreground

File: styles/mixins/tooltip.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        'position': 'absolute',
+        'z-index': 10,
+        'opacity': 0,
+        'border-radius': '4px',
+        'padding': '.5em 1em',
+        'background-color': color,
+        'color': '#fff',
+        'font-size': '12px',
+        'white-space': 'nowrap',
+        'content': 'attr(data-tooltip-title)',
```
Comment (jgallen23) [inline styles/mixins/tooltip.js]: this work everywhere?

File: styles/mixins/tooltip.js (hunk 4 above)
Comment (jgallen23) [inline styles/mixins/tooltip.js]: any chance we can get direction passed in here? if no direction passed, then loop through and set everything. That way we can be consistent with our other mixins

**Comment by jgallen23** [comment]
can you add to examples/index.html?

File: styles/mixins/tooltip.js (hunk 5 above)
Comment (chrisopedia) [inline styles/mixins/tooltip.js]: https://developer.mozilla.org/en-US/docs/Web/CSS/attr#Browser_compatibility

- chrome 2.0
- firefox 1.0
- IE 8
- Safari 3.1

File: styles/mixins/tooltip.js (hunk 4 above)
Comment (chrisopedia) [inline styles/mixins/tooltip.js]: The only issue was there was a lot of complex nested and duplicate selectors and I wasn't sure how to do the mixins like that.  For example:

https://github.com/firstandthird/clientkit/pull/39/files#diff-cb4b362339a1552ea99c0267018a7fd6R126

Comment (jgallen23) [inline styles/mixins/tooltip.js]: you just return a JavaScript object. it can be nested

**Comment by jgallen23** [comment]
still need to add to examples/index.html and merge in latest from master

**Comment by jgallen23** [comment]
ping

**Comment by chrisopedia** [comment]
Updated to include master but that introduced build errors.  Also added in the examples.

**Comment by jgallen23** [comment]
can't merge

**Comment by chrisopedia** [comment]
@jgallen23 done

**Comment by jgallen23** [comment]
lets update this to the new standard

**Comment by chrisopedia** [comment]
@jgallen23 done

**Pull request merged**